src/CMakeLists.txt

```diff
@@ -45,29 +56,32 @@ add_subdirectory(${CLM_ROOT}/src/self_tests clm_self_tests)
 add_subdirectory(unit_test_stubs)
 add_subdirectory(unit_test_shr)
 
-# Remove shr_mpi_mod from share_sources.
-# This is needed because we want to use the mock shr_mpi_mod in place of the real one
+# Remove some things from share_sources
 #
 # TODO: this should be moved into a general-purpose function in Sourcelist_utils.
-# Then this block of code could be replaced with a single call, like:
+# Then each removal could be replaced with a single call, like:
 # remove_source_file(${share_sources} "shr_mpi_mod.F90")
 foreach (sourcefile ${share_sources})
+  # Remove shr_mpi_mod from share_sources.
+  # This is needed because we want to use the mock shr_mpi_mod in place of the real one
   string(REGEX MATCH "shr_mpi_mod.F90" match_found ${sourcefile})
   if(match_found)
     list(REMOVE_ITEM share_sources ${sourcefile})
   endif()
-endforeach()
 
-# Bring in PIO, jsut because it's needed for the default ESMF library and included in other submodules like share and cmeps
-if (DEFINED PIO)
-  set(PIO_PATH ${PIO})
-else()
-  set(PIO_PATH $ENV{PIO})
-endif()
+  # Remove shr_pio_mod from share_sources. This is needed to avoid an explicit dependency
```

Comment:
Hmmm, since the following isn't required -- I wonder about removing it? It does seem like it's reasonable to expect that the builds of ESMF used for PFunit testing have PIO built in -- right?

It's good for us to use code in shr_pio_mod when we can, so I'd rather have it included, unless it's really necessary to remove.

Reply:
In talking this over we are leaving this in, because it would complicate the build for supporting ESMF with internal PIO. Longer term we should maybe require having ESMF built with pre-built PIO, but this complicates the CESM build. And as I/O is something we try to avoid in unit testing it makes sense to just do this.